src/assets/wise5/authoringTool/peer-grouping/edit-peer-grouping-dialog/edit-peer-grouping-dialog.component.ts

```diff
@@ -25,14 +31,63 @@ export class EditPeerGroupingDialogComponent extends AuthorPeerGroupingDialogCom
   ngOnInit(): void {
     this.peerGrouping = this.utilService.makeCopyOfJSONObject(this.peerGrouping);
     this.stepsUsedIn = this.peerGroupingAuthoringService.getStepsUsedIn(this.peerGrouping.tag);
+    this.logicType = this.getLogicType(this.peerGrouping.logic);
+    if (this.logicType === DIFFERENT_IDEAS_VALUE) {
+      this.referenceComponent = this.getDifferentIdeasReferenceComponent(this.peerGrouping.logic);
+    }
+  }
+
+  private getLogicType(logic: string): string {
+    if (new RegExp(DIFFERENT_IDEAS_REGEX).exec(logic) != null) {
+      return DIFFERENT_IDEAS_VALUE;
+    } else {
+      return logic;
+    }
+  }
+
+  private getDifferentIdeasReferenceComponent(logic: string): any {
+    const result = new RegExp(DIFFERENT_IDEAS_REGEX).exec(logic);
+    return {
+      nodeId: result[1],
+      componentId: result[2]
+    };
+  }
+
+  referenceComponentNodeIdChanged(event: any): void {
+    let numAllowedComponents = 0;
+    let allowedComponent = null;
+    for (const component of this.projectService.getComponentsByNodeId(event.nodeId)) {
+      if (this.allowedReferenceComponentTypes.includes(component.type)) {
+        numAllowedComponents += 1;
+        allowedComponent = component;
+      }
+    }
+    if (numAllowedComponents === 1) {
+      this.referenceComponent.componentId = allowedComponent.id;
+    } else {
+      this.referenceComponent.componentId = null;
+    }
   }
 
   save(): void {
+    this.injectPeerGroupingLogic(this.peerGrouping);
     this.peerGroupingAuthoringService.updatePeerGrouping(this.peerGrouping).subscribe(() => {
       this.dialogRef.close();
     });
   }
 
+  private injectPeerGroupingLogic(peerGrouping: PeerGrouping): void {
+    if (this.logicType === DIFFERENT_IDEAS_VALUE) {
+      peerGrouping.logic = this.generateDifferentIdeasLogic();
+    } else {
+      peerGrouping.logic = this.logicType;
+    }
```

Comment:
Use ternary to reduce complexity and verbosity? Also generate differentIdeasLogic string in this function? I also wonder if this function should be called updatePeerGroupingLogic() without taking in a parameter..."inject" doesn't feel like the right verb; it's more like we're setting/updating the PeerGrouping.logic field.
```suggestion
    this.peerGrouping.logic =
      this.logicType === DIFFERENT_IDEAS_VALUE
        ? `differentIdeas("${this.referenceComponent.nodeId}", "${this.referenceComponent.componentId}")`
        : this.logicType;
```